pkg/middleware/cached.go

```diff
@@ -87,26 +87,28 @@ func (mw *CachedOptlyMiddleware) ClientCtx(next http.Handler) http.Handler {
 // detail from a UPS or attribute store.
 func (mw *CachedOptlyMiddleware) UserCtx(next http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-
 		userID := chi.URLParam(r, "userID")
 		if userID == "" {
 			RenderError(fmt.Errorf("invalid request, missing userId"), http.StatusBadRequest, w, r)
 			return
 		}
 
 		// Remove userId and copy values into the attributes map
+		logger := GetLogger(r)
 		values := r.URL.Query()
 		attributes := make(map[string]interface{})
+
 		for k, v := range values {
 			// Assuming a single KV pair exists in the query parameters
-			attributes[k] = v[0]
-			GetLogger(r).Debug().Str("attrKey", k).Str("attrVal", v[0]).Msg("User attribute.")
+			val := v[0]
+			attributes[k] = CoerceType(val)
```

Comment:
I thought we will be providing an option to do that, not all the time.

Reply:
This is for attributes in the request we should always coerce these. The optional typing is on the response for feature variables which will be part of a different PR.